### PR TITLE
Auto generate docs for javascript

### PR DESCRIPTION
**Pull request by rpunkfu**

### CODE REVIEW

**Comment by zzarcon** [comment]
@ryyppy I think you will love this ;)

**Comment by rpunkfu** [comment]
Implements #21

File: package.json [hunk 1]
```diff
@@ -40,7 +47,7 @@
   },
   "config": {
     "ghooks": {
-      "pre-commit": "eslint gh-emoji.js"
+      "pre-commit": "npm run pre-commit"
```
Comment (ryyppy) [inline package.json:53]: yes, I really like this!

**Comment by ryyppy** [comment]
Looks great! We should wait for PR #14 before merging this! I am on it

**Comment by rpunkfu** [comment]
I will resolve #24 in this PR in a sec

**Comment by rpunkfu** [comment]
Now this PR resolves both #21 and #24

**Comment by zzarcon** [comment]
So, is this ready to merge? @oskarcieslik @ryyppy

**Comment by rpunkfu** [comment]
lgtm :muscle:

**Comment by ryyppy** [comment]
I wonder, don't we need a documentationjs config file? everything is just running with the default config?

**Comment by rpunkfu** [comment]
Does configuration file give any functionality other than order of documentation content?

**Comment by rpunkfu** [comment]
Btw, I think that methods should also be stored in alphabetical order, just like in our generated API

**Pull request merged**